HLTrigger/btau/src/HLTDisplacedtktktkVtxProducer.cc

```diff
@@ -0,0 +1,280 @@
+  srcTag_ (iConfig.getParameter<edm::InputTag>("Src")),
+  srcToken_(consumes<reco::RecoChargedCandidateCollection>(srcTag_)),
+  previousCandTag_(iConfig.getParameter<edm::InputTag>("PreviousCandTag")),
+  previousCandToken_(consumes<trigger::TriggerFilterObjectWithRefs>(previousCandTag_)),
+  maxEta_ (iConfig.getParameter<double>("MaxEtaTk")),
+  minPtTk1_ (iConfig.getParameter<double>("MinPtResTk1")),
+  minPtTk2_ (iConfig.getParameter<double>("MinPtResTk2")),
+  minPtTk3_ (iConfig.getParameter<double>("MinPtThirdTk")),
+  minPtRes_ (iConfig.getParameter<double>("MinPtRes")),
+  minPtTri_ (iConfig.getParameter<double>("MinPtTri")),
+  minInvMassRes_ (iConfig.getParameter<double>("MinInvMassRes")),
+  maxInvMassRes_ (iConfig.getParameter<double>("MaxInvMassRes")),
+  minInvMass_ (iConfig.getParameter<double>("MinInvMass")),
+  maxInvMass_ (iConfig.getParameter<double>("MaxInvMass")),
+  massParticle1_ (iConfig.getParameter<double>("massParticleRes1")),
+  massParticle2_ (iConfig.getParameter<double>("massParticleRes2")),
+  massParticle3_ (iConfig.getParameter<double>("massParticle3")),
+  chargeOpt_ (iConfig.getParameter<int>("ChargeOpt")),
+  resOpt_ (iConfig.getParameter<int>("ResOpt")),
+  triggerTypeDaughters_(iConfig.getParameter<int>("triggerTypeDaughters"))
+
+{
+  produces<VertexCollection>();
+}
+
+
+HLTDisplacedtktktkVtxProducer::~HLTDisplacedtktktkVtxProducer() = default;
+
+void
+HLTDisplacedtktktkVtxProducer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("Src",edm::InputTag("hltL3MuonCandidates"));
+  desc.add<edm::InputTag>("PreviousCandTag",edm::InputTag(""));
+  desc.add<double>("MaxEtaTk",2.5);
+  desc.add<double>("MinPtResTk1",0.0);
+  desc.add<double>("MinPtResTk2",0.0);
+  desc.add<double>("MinPtThirdTk",0.0);
+  desc.add<double>("MinPtRes",0.0);
+  desc.add<double>("MinPtTri",0.0);
+  desc.add<double>("MinInvMassRes",1.0);
+  desc.add<double>("MaxInvMassRes",20.0);
+  desc.add<double>("MinInvMass",1.0);
+  desc.add<double>("MaxInvMass",20.0);
+  desc.add<double>("massParticleRes1",0.4937);
+  desc.add<double>("massParticleRes2",0.4937);
+  desc.add<double>("massParticle3",0.1396);
+  desc.add<int>("ChargeOpt",-1);
+  desc.add<int>("ResOpt",1);
+  desc.add<int>("triggerTypeDaughters",0);
+
+  descriptions.add("hltDisplacedtktktkVtxProducer", desc);
+}
+
+// ------------ method called once each job just before starting event loop  ------------
+void HLTDisplacedtktktkVtxProducer::beginJob()
+{
+
+}
+
+// ------------ method called once each job just after ending the event loop  ------------
+void HLTDisplacedtktktkVtxProducer::endJob() 
+{
+ 	
+}
+
+// ------------ method called on each new Event  ------------
+void HLTDisplacedtktktkVtxProducer::produce(edm::Event& iEvent, const edm::EventSetup& iSetup)
+{
+  double firstTrackMass = massParticle1_;
```

Comment:
these next 20 lines can all happen once in the constructor, right?

Reply:
changed